src/fetchData/fetchPageData.ts

```diff
@@ -115,10 +116,11 @@ export default async function fetchPageData({
   }
 
   if (page.getServerSideProps) {
+    const { getServerSideProps } = page;
     const ctx: GetServerSidePropsContext<typeof params> = makeGetServerSidePropsContext(
       { options, pageObject }
     );
-    const pageData = await page.getServerSideProps(ctx);
+    const pageData = await executeAsIfOnServer(() => getServerSideProps(ctx));
```

Comment:
`getServerSideProps` always executes on server. For libraries are checking for `window` and `document` existence. For them to work as expected, we need to execute it in such context.